www/js/metrics/metricsHelper.ts

```diff
@@ -106,131 +89,16 @@ export function calculatePercentChange(pastWeekRange, previousWeekRange) {
-        /*sum all possible on foot modes: see https://github.com/e-mission/e-mission-docs/issues/422 */
-        if (isOnFoot(field)) {
-          onFootVal += metricToValue(population, metric, field);
-          field = 'ON_FOOT';
-        }
-        if (!(field in mode_bins)) {
-          mode_bins[field] = [];
-        }
-        //for all except onFoot, add to bin - could discover mult onFoot modes
-        if (field != 'ON_FOOT') {
-          mode_bins[field].push([
-            metric.ts,
-            metricToValue(population, metric, field),
-            metric.fmt_time,
-          ]);
-        }
-      }
-      const trimmedField = trimGroupingPrefix(field);
-      if (trimmedField) {
-        logDebug('Mapped field ' + field + ' to mode ' + trimmedField);
-        if (!(trimmedField in mode_bins)) {
-          mode_bins[trimmedField] = [];
-        }
-        mode_bins[trimmedField].push([
-          metric.ts,
-          Math.round(metricToValue(population, metric, field)),
-          DateTime.fromISO(metric.fmt_time).toISO() as string,
-        ]);
-      }
-    }
-    //handle the ON_FOOT modes once all have been summed
-    if ('ON_FOOT' in mode_bins) {
-      mode_bins['ON_FOOT'].push([metric.ts, Math.round(onFootVal), metric.fmt_time]);
-    }
-  });
-
-  return Object.entries(mode_bins).map(([key, values]) => ({ key, values }));
-}
-
 const _datesTsCache = {};
 export const tsForDayOfMetricData = (day: DayOfMetricData) => {
   if (_datesTsCache[day.date] == undefined)
     _datesTsCache[day.date] = DateTime.fromISO(day.date).toSeconds();
   return _datesTsCache[day.date];
 };
 
-export const valueForFieldOnDay = (day: DayOfMetricData, field: string, key: string) =>
+export const valueForFieldOnDay = (day: MetricEntry, field: string, key: string) =>
   day[`${field}_${key}`];
 
-export type MetricsSummary = { key: string; values: number };
-export function generateSummaryFromData(modeMap, metric) {
-  logDebug(`Invoked getSummaryDataRaw on ${JSON.stringify(modeMap)} with ${metric}`);
-
-  let summaryMap: MetricsSummary[] = [];
-
-  for (let i = 0; i < modeMap.length; i++) {
-    let vals = 0;
-    for (let j = 0; j < modeMap[i].values.length; j++) {
-      vals += modeMap[i].values[j][1]; //2nd item of array is value
-    }
-    if (metric === 'mean_speed') {
-      // For speed, we take the avg. For other metrics we keep the sum
-      vals = vals / modeMap[i].values.length;
-    }
-    summaryMap.push({
-      key: modeMap[i].key,
-      values: Math.round(vals),
-    });
-  }
-
-  return summaryMap;
-}
-
-/*
- * We use the results to determine whether these results are from custom
- * labels or from the automatically sensed labels. Automatically sensedV
- * labels are in all caps, custom labels are prefixed by label, but have had
- * the label_prefix stripped out before this. Results should have either all
- * sensed labels or all custom labels.
- */
-export function isCustomLabels(modeMap) {
```

Comment:
I don't think this has been supported in the UI for a while (likely since the UI migration).

> some community members do so, as seen in e-mission/e-mission-server#960

Good to know, I was not aware anyone was using non-"custom"

I do have a plan that will allow sensed or inferred modes to be used, depending on the dynamic config metrics.phone_dashboard_ui.metric_list

It defaults to
```
{
  footprint: ['mode_confirm'],
  distance: ['mode_confirm'],
  duration: ['mode_confirm'],
  count: ['mode_confirm'],
}
```

we can easily add `purpose_confirm`
```
{
  footprint: ['mode_confirm', 'purpose_confirm'],
  distance: ['mode_confirm', 'purpose_confirm'],
  duration: ['mode_confirm', 'purpose_confirm'],
  count: ['mode_confirm', 'purpose_confirm'],
}
```

In the future, maybe something like
```
{
  footprint: ['primary_inferred_mode'],
  distance: ['primary_inferred_mode'],
  duration: ['primary_inferred_mode'],
  count: ['primary_inferred_mode'],
}
```

or just
```
{
  footprint: ['primary_mode'],
  distance: ['primary_mode'],
  duration: ['primary_mode'],
  count: ['primary_mode'],
}
```

Adding a new one of these "grouping fields" is trivial https://github.com/JGreenlee/e-mission-common/blob/52af159cfb3dd6a1cd4c8ba37ba4bc057ca24ca4/src/emcommon/metrics/metrics_summaries.py#L132-L140

The shortcoming of this plan, however, is that all calculations are still be on a per-trip basis, not a per-section basis. It would use the primary mode and non-primary modes would be ignored.
Per-section calculations would require significant changes.